tests/unit/test_charm.py

```diff
@@ -508,211 +418,138 @@ def test_given_config_files_and_relations_are_created_when_configure_sdcore_smf_
         )
 
-    @patch(f"{CERTIFICATES_LIB}.request_certificate_creation")
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    @patch("charm.check_output")
     def test_given_cert_already_stored_when_on_certificates_relation_joined_then_cert_is_not_requested(  # noqa: E501
-        self, patch_check_output, patched_nrf_url, patch_request_certificate_creation
+        self, add_storage, mock_default_values
     ):
-        self.harness.add_storage(storage_name="certs", attach=True)
-        self.harness.add_storage(storage_name="config", attach=True)
         root = self.harness.get_filesystem_root(self.container_name)
-        (root / "support/TLS/smf.key").write_text(PRIVATE_KEY)
-        (root / "support/TLS/smf.pem").write_text(CERTIFICATE)
+        (root / PRIVATE_KEY_PATH).write_text(PRIVATE_KEY)
+        (root / CERTIFICATE_PATH).write_text(CERTIFICATE)
         self.harness.set_can_connect(container=self.container_name, val=True)
-        patch_check_output.return_value = POD_IP
-        patched_nrf_url.return_value = VALID_NRF_URL
 
-        patch_request_certificate_creation.assert_not_called()
+        self.mock_request_certificate_creation.assert_not_called()
 
-    @patch(f"{CERTIFICATES_LIB}.get_assigned_certificates")
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    @patch("charm.check_output")
     def test_given_csr_matches_stored_one_when_certificate_available_then_certificate_is_pushed(
-        self, patch_check_output, patched_nrf_url, patch_get_assigned_certificates
+        self, add_storage, nrf_relation_id, certificates_relation_id, mock_default_values
     ):
-        self.harness.add_storage(storage_name="certs", attach=True)
-        self.harness.add_storage(storage_name="config", attach=True)
         root = self.harness.get_filesystem_root(self.container_name)
-        (root / "support/TLS/smf.key").write_text(PRIVATE_KEY)
-        (root / "support/TLS/smf.csr").write_text(CSR)
-        (root / "etc/smf/uerouting.yaml").write_text(self._read_file("src/uerouting.yaml"))
-        patch_check_output.return_value = POD_IP
-        patched_nrf_url.return_value = VALID_NRF_URL
+        (root / PRIVATE_KEY_PATH).write_text(PRIVATE_KEY)
+        (root / CSR_PATH).write_text(CSR)
+        (root / UE_CONFIG_FILE_PATH).write_text(self._read_file(EXPECTED_UE_CONFIG_FILE_PATH))
         self._create_database_relation_and_populate_data()
-        self._create_nrf_relation()
-        self._create_certificates_relation()
 
-        provider_certificate = Mock(ProviderCertificate)
-        provider_certificate.certificate = CERTIFICATE
-        provider_certificate.csr = CSR
-        patch_get_assigned_certificates.return_value = [provider_certificate]
+        self.mock_get_assigned_certificates.return_value = [self._get_provider_certificate()]
         self.harness.set_can_connect(container=self.container_name, val=True)
```

Comment:
```suggestion
```
Pebble ready sets can connect to True so this line can be removed

Reply:
Done